In [1]:
from google.colab import files

In [2]:
uploaded = files.upload()

Saving train.csv to train.csv


In [3]:
#pip install polars
#pip instal pandas

In [4]:
import numpy as np
import polars as pl
import pandas as pd

In [5]:
df_pl = pl.read_csv('train.csv')

In [6]:
df_pl.describe()

statistic,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
str,f64,f64,f64,str,str,f64,f64,f64,str,f64,str,str
"""count""",891.0,891.0,891.0,"""891""","""891""",714.0,891.0,891.0,"""891""",891.0,"""204""","""889"""
"""null_count""",0.0,0.0,0.0,"""0""","""0""",177.0,0.0,0.0,"""0""",0.0,"""687""","""2"""
"""mean""",446.0,0.383838,2.308642,null,null,29.699118,0.523008,0.381594,null,32.204208,null,null
"""std""",257.353842,0.486592,0.836071,null,null,14.526497,1.102743,0.806057,null,49.693429,null,null
"""min""",1.0,0.0,1.0,"""Abbing, Mr. Anthony""","""female""",0.42,0.0,0.0,"""110152""",0.0,"""A10""","""C"""
"""25%""",224.0,0.0,2.0,null,null,20.0,0.0,0.0,null,7.925,null,null
"""50%""",446.0,0.0,3.0,null,null,28.0,0.0,0.0,null,14.4542,null,null
"""75%""",669.0,1.0,3.0,null,null,38.0,1.0,0.0,null,31.0,null,null
"""max""",891.0,1.0,3.0,"""van Melkebeke, Mr. Philemon""","""male""",80.0,8.0,6.0,"""WE/P 5735""",512.3292,"""T""","""S"""


In [7]:
df_pl.mean()

PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
f64,f64,f64,str,str,f64,f64,f64,str,f64,str,str
446.0,0.383838,2.308642,null,null,29.699118,0.523008,0.381594,null,32.204208,null,null


In [8]:
df_pl.dtypes

[Int64,
 Int64,
 Int64,
 String,
 String,
 Float64,
 Int64,
 Int64,
 String,
 Float64,
 String,
 String]

In [9]:
df_pl.null_count()

PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
u32,u32,u32,u32,u32,u32,u32,u32,u32,u32,u32,u32
0,0,0,0,0,177,0,0,0,0,687,2


In [10]:
df_pl.height

891

In [11]:
df_pl.width

12

In [12]:
df_pl.shape

(891, 12)

In [13]:
df_pl.columns

['PassengerId',
 'Survived',
 'Pclass',
 'Name',
 'Sex',
 'Age',
 'SibSp',
 'Parch',
 'Ticket',
 'Fare',
 'Cabin',
 'Embarked']

In [14]:
df_pl.schema

Schema([('PassengerId', Int64),
        ('Survived', Int64),
        ('Pclass', Int64),
        ('Name', String),
        ('Sex', String),
        ('Age', Float64),
        ('SibSp', Int64),
        ('Parch', Int64),
        ('Ticket', String),
        ('Fare', Float64),
        ('Cabin', String),
        ('Embarked', String)])

In [15]:
df_pl.is_empty()

False

In [16]:
df_pl.head(15)

PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
i64,i64,i64,str,str,f64,i64,i64,str,f64,str,str
1,0,3,"""Braund, Mr. Owen Harris""","""male""",22.0,1,0,"""A/5 21171""",7.25,null,"""S"""
2,1,1,"""Cumings, Mrs. John Bradley (Fl…","""female""",38.0,1,0,"""PC 17599""",71.2833,"""C85""","""C"""
3,1,3,"""Heikkinen, Miss. Laina""","""female""",26.0,0,0,"""STON/O2. 3101282""",7.925,null,"""S"""
4,1,1,"""Futrelle, Mrs. Jacques Heath (…","""female""",35.0,1,0,"""113803""",53.1,"""C123""","""S"""
5,0,3,"""Allen, Mr. William Henry""","""male""",35.0,0,0,"""373450""",8.05,null,"""S"""
…,…,…,…,…,…,…,…,…,…,…,…
11,1,3,"""Sandstrom, Miss. Marguerite Ru…","""female""",4.0,1,1,"""PP 9549""",16.7,"""G6""","""S"""
12,1,1,"""Bonnell, Miss. Elizabeth""","""female""",58.0,0,0,"""113783""",26.55,"""C103""","""S"""
13,0,3,"""Saundercock, Mr. William Henry""","""male""",20.0,0,0,"""A/5. 2151""",8.05,null,"""S"""


In [17]:
df_pl.tail(101)

PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
i64,i64,i64,str,str,f64,i64,i64,str,f64,str,str
791,0,3,"""Keane, Mr. Andrew ""Andy""""","""male""",null,0,0,"""12460""",7.75,null,"""Q"""
792,0,2,"""Gaskell, Mr. Alfred""","""male""",16.0,0,0,"""239865""",26.0,null,"""S"""
793,0,3,"""Sage, Miss. Stella Anna""","""female""",null,8,2,"""CA. 2343""",69.55,null,"""S"""
794,0,1,"""Hoyt, Mr. William Fisher""","""male""",null,0,0,"""PC 17600""",30.6958,null,"""C"""
795,0,3,"""Dantcheff, Mr. Ristiu""","""male""",25.0,0,0,"""349203""",7.8958,null,"""S"""
…,…,…,…,…,…,…,…,…,…,…,…
887,0,2,"""Montvila, Rev. Juozas""","""male""",27.0,0,0,"""211536""",13.0,null,"""S"""
888,1,1,"""Graham, Miss. Margaret Edith""","""female""",19.0,0,0,"""112053""",30.0,"""B42""","""S"""
889,0,3,"""Johnston, Miss. Catherine Hele…","""female""",null,1,2,"""W./C. 6607""",23.45,null,"""S"""


In [18]:
df_pl.unique()

PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
i64,i64,i64,str,str,f64,i64,i64,str,f64,str,str
509,0,3,"""Olsen, Mr. Henry Margido""","""male""",28.0,0,0,"""C 4001""",22.525,null,"""S"""
713,1,1,"""Taylor, Mr. Elmer Zebley""","""male""",48.0,1,0,"""19996""",52.0,"""C126""","""S"""
428,1,2,"""Phillips, Miss. Kate Florence …","""female""",19.0,0,0,"""250655""",26.0,null,"""S"""
790,0,1,"""Guggenheim, Mr. Benjamin""","""male""",46.0,0,0,"""PC 17593""",79.2,"""B82 B84""","""C"""
763,1,3,"""Barah, Mr. Hanna Assi""","""male""",20.0,0,0,"""2663""",7.2292,null,"""C"""
…,…,…,…,…,…,…,…,…,…,…,…
473,1,2,"""West, Mrs. Edwy Arthur (Ada Ma…","""female""",33.0,1,2,"""C.A. 34651""",27.75,null,"""S"""
503,0,3,"""O'Sullivan, Miss. Bridget Mary""","""female""",null,0,0,"""330909""",7.6292,null,"""Q"""
303,0,3,"""Johnson, Mr. William Cahoone J…","""male""",19.0,0,0,"""LINE""",0.0,null,"""S"""


In [19]:
df_pl.fill_null

<bound method DataFrame.fill_null of shape: (891, 12)
┌─────────────┬──────────┬────────┬──────────────────┬───┬────────────┬─────────┬───────┬──────────┐
│ PassengerId ┆ Survived ┆ Pclass ┆ Name             ┆ … ┆ Ticket     ┆ Fare    ┆ Cabin ┆ Embarked │
│ ---         ┆ ---      ┆ ---    ┆ ---              ┆   ┆ ---        ┆ ---     ┆ ---   ┆ ---      │
│ i64         ┆ i64      ┆ i64    ┆ str              ┆   ┆ str        ┆ f64     ┆ str   ┆ str      │
╞═════════════╪══════════╪════════╪══════════════════╪═══╪════════════╪═════════╪═══════╪══════════╡
│ 1           ┆ 0        ┆ 3      ┆ Braund, Mr. Owen ┆ … ┆ A/5 21171  ┆ 7.25    ┆ null  ┆ S        │
│             ┆          ┆        ┆ Harris           ┆   ┆            ┆         ┆       ┆          │
│ 2           ┆ 1        ┆ 1      ┆ Cumings, Mrs.    ┆ … ┆ PC 17599   ┆ 71.2833 ┆ C85   ┆ C        │
│             ┆          ┆        ┆ John Bradley     ┆   ┆            ┆         ┆       ┆          │
│             ┆          ┆        ┆ (Fl…             ┆   ┆            ┆         ┆       ┆          │
│ 3           ┆ 1        ┆ 3      ┆ Heikkinen, Miss. ┆ … ┆ STON/O2.   ┆ 7.925   ┆ null  ┆ S        │
│             ┆          ┆        ┆ Laina            ┆   ┆ 3101282    ┆         ┆       ┆          │
│ 4           ┆ 1        ┆ 1      ┆ Futrelle, Mrs.   ┆ … ┆ 113803     ┆ 53.1    ┆ C123  ┆ S        │
│             ┆          ┆        ┆ Jacques Heath (… ┆   ┆            ┆         ┆       ┆          │
│ 5           ┆ 0        ┆ 3      ┆ Allen, Mr.       ┆ … ┆ 373450     ┆ 8.05    ┆ null  ┆ S        │
│             ┆          ┆        ┆ William Henry    ┆   ┆            ┆         ┆       ┆          │
│ …           ┆ …        ┆ …      ┆ …                ┆ … ┆ …          ┆ …       ┆ …     ┆ …        │
│ 887         ┆ 0        ┆ 2      ┆ Montvila, Rev.   ┆ … ┆ 211536     ┆ 13.0    ┆ null  ┆ S        │
│             ┆          ┆        ┆ Juozas           ┆   ┆            ┆         ┆       ┆          │
│ 888         ┆ 1        ┆ 1      ┆ Graham, Miss.    ┆ … ┆ 112053     ┆ 30.0    ┆ B42   ┆ S        │
│             ┆          ┆        ┆ Margaret Edith   ┆   ┆            ┆         ┆       ┆          │
│ 889         ┆ 0        ┆ 3      ┆ Johnston, Miss.  ┆ … ┆ W./C. 6607 ┆ 23.45   ┆ null  ┆ S        │
│             ┆          ┆        ┆ Catherine Hele…  ┆   ┆            ┆         ┆       ┆          │
│ 890         ┆ 1        ┆ 1      ┆ Behr, Mr. Karl   ┆ … ┆ 111369     ┆ 30.0    ┆ C148  ┆ C        │
│             ┆          ┆        ┆ Howell           ┆   ┆            ┆         ┆       ┆          │
│ 891         ┆ 0        ┆ 3      ┆ Dooley, Mr.      ┆ … ┆ 370376     ┆ 7.75    ┆ null  ┆ Q        │
│             ┆          ┆        ┆ Patrick          ┆   ┆            ┆         ┆       ┆          │
└─────────────┴──────────┴────────┴──────────────────┴───┴────────────┴─────────┴───────┴──────────┘>

In [20]:
df_pl['Pclass'].value_counts()

Pclass,count
i64,u32
2,184
1,216
3,491


In [21]:
df_pl['Survived'].value_counts()

Survived,count
i64,u32
1,342
0,549


In [22]:
df_pl['Sex'].value_counts()

Sex,count
str,u32
"""female""",314
"""male""",577


In [168]:
df_pl.group_by("Sex").agg(pl.col("Survived").sum())

Sex,Survived
str,i64
"""female""",233
"""male""",109


In [169]:
df_pl.group_by("Sex").agg(pl.col("Survived").mean())

Sex,Survived
str,f64
"""male""",0.188908
"""female""",0.742038


In [25]:
df_pl.filter(pl.col('Age') > 44)

PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
i64,i64,i64,str,str,f64,i64,i64,str,f64,str,str
7,0,1,"""McCarthy, Mr. Timothy J""","""male""",54.0,0,0,"""17463""",51.8625,"""E46""","""S"""
12,1,1,"""Bonnell, Miss. Elizabeth""","""female""",58.0,0,0,"""113783""",26.55,"""C103""","""S"""
16,1,2,"""Hewlett, Mrs. (Mary D Kingcome…","""female""",55.0,0,0,"""248706""",16.0,null,"""S"""
34,0,2,"""Wheadon, Mr. Edward H""","""male""",66.0,0,0,"""C.A. 24579""",10.5,null,"""S"""
53,1,1,"""Harper, Mrs. Henry Sleeper (My…","""female""",49.0,1,0,"""PC 17572""",76.7292,"""D33""","""C"""
…,…,…,…,…,…,…,…,…,…,…,…
858,1,1,"""Daly, Mr. Peter Denis ""","""male""",51.0,0,0,"""113055""",26.55,"""E17""","""S"""
863,1,1,"""Swift, Mrs. Frederick Joel (Ma…","""female""",48.0,0,0,"""17466""",25.9292,"""D17""","""S"""
872,1,1,"""Beckwith, Mrs. Richard Leonard…","""female""",47.0,1,1,"""11751""",52.5542,"""D35""","""S"""


In [26]:
df_pd = pd.read_csv('train.csv')

In [27]:
!pip install bottleneck

In [28]:
import bottleneck as bn

In [29]:
bn.nanmean(df_pd['Age'])

29.69911764705882

In [30]:
bn.nanstd(df_pd['Age'])

14.516321150817317

In [31]:
df_pd['Fare_new'] = df_pd['Fare'].apply(lambda row: row * 1.3)

In [32]:
df_pd.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked,Fare_new
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S,9.42500
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C,92.66829
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S,10.30250
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S,69.03000
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S,10.46500


In [139]:
from google.colab import files
uploaded = files.upload()


Saving Housing.csv to Housing (2).csv


In [140]:
df_housing = pd.read_csv('Housing.csv')

In [141]:
df_housing.head()

,price,area,bedrooms,bathrooms,stories,mainroad,guestroom,basement,hotwaterheating,airconditioning,parking,prefarea,furnishingstatus
0,13300000,7420,4,2,3,yes,no,no,no,yes,2,yes,furnished
1,12250000,8960,4,4,4,yes,no,no,no,yes,3,no,furnished
2,12250000,9960,3,2,2,yes,no,yes,no,no,2,yes,semi-furnished
3,12215000,7500,4,2,2,yes,no,yes,no,yes,3,yes,furnished
4,11410000,7420,4,1,2,yes,yes,yes,no,yes,2,no,furnished


In [142]:
df_housing.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 545 entries, 0 to 544
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   price             545 non-null    int64 
 1   area              545 non-null    int64 
 2   bedrooms          545 non-null    int64 
 3   bathrooms         545 non-null    int64 
 4   stories           545 non-null    int64 
 5   mainroad          545 non-null    object
 6   guestroom         545 non-null    object
 7   basement          545 non-null    object
 8   hotwaterheating   545 non-null    object
 9   airconditioning   545 non-null    object
 10  parking           545 non-null    int64 
 11  prefarea          545 non-null    object
 12  furnishingstatus  545 non-null    object
dtypes: int64(6), object(7)
memory usage: 55.5+ KB


**Выводы:**

Для оптимизации типы столбцов price, bedrooms, bathrooms, stories и parking можно изменить с int64 на int8. Тип столбца area на int16.
Столбцы mainroad, guestroom, basement, hotwaterheating, airconditioning, prefarea и furnishingstatus содержат строковые данные; для уменьшения занимаемой памяти их следует преобразовать из типа object в category.


In [143]:
df_housing.describe()

,price,area,bedrooms,bathrooms,stories,parking
count,5.450000e+02,545.000000,545.000000,545.000000,545.000000,545.000000
mean,4.766729e+06,5150.541284,2.965138,1.286239,1.805505,0.693578
std,1.870440e+06,2170.141023,0.738064,0.502470,0.867492,0.861586
min,1.750000e+06,1650.000000,1.000000,1.000000,1.000000,0.000000
25%,3.430000e+06,3600.000000,2.000000,1.000000,1.000000,0.000000
50%,4.340000e+06,4600.000000,3.000000,1.000000,2.000000,0.000000
75%,5.740000e+06,6360.000000,3.000000,2.000000,2.000000,1.000000
max,1.330000e+07,16200.000000,6.000000,4.000000,4.000000,3.000000


In [144]:
df_housing = df_housing.fillna(0)
df_housing['price'] = df_housing['price'].astype(np.int8)
df_housing['area'] = df_housing['area'].astype(np.int16)
df_housing['bedrooms'] = df_housing['bedrooms'].astype(np.int8)
df_housing['bathrooms'] = df_housing['bathrooms'].astype(np.int8)
df_housing['stories'] = df_housing['stories'].astype(np.int8)
df_housing['parking'] = df_housing['parking'].astype(np.int8)


In [145]:
df_housing.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 545 entries, 0 to 544
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   price             545 non-null    int8  
 1   area              545 non-null    int16 
 2   bedrooms          545 non-null    int8  
 3   bathrooms         545 non-null    int8  
 4   stories           545 non-null    int8  
 5   mainroad          545 non-null    object
 6   guestroom         545 non-null    object
 7   basement          545 non-null    object
 8   hotwaterheating   545 non-null    object
 9   airconditioning   545 non-null    object
 10  parking           545 non-null    int8  
 11  prefarea          545 non-null    object
 12  furnishingstatus  545 non-null    object
dtypes: int16(1), int8(5), object(7)
memory usage: 33.7+ KB


In [146]:
df_housing['mainroad'].info(memory_usage="deep")

<class 'pandas.core.series.Series'>
RangeIndex: 545 entries, 0 to 544
Series name: mainroad
Non-Null Count  Dtype 
--------------  ----- 
545 non-null    object
dtypes: object(1)
memory usage: 27.7 KB


In [147]:
df_housing['mainroad'] = df_housing['mainroad'].astype("category")
df_housing['mainroad'].info(memory_usage="deep")


<class 'pandas.core.series.Series'>
RangeIndex: 545 entries, 0 to 544
Series name: mainroad
Non-Null Count  Dtype   
--------------  -----   
545 non-null    category
dtypes: category(1)
memory usage: 888.0 bytes


In [148]:
df_housing['guestroom'].info(memory_usage="deep")

<class 'pandas.core.series.Series'>
RangeIndex: 545 entries, 0 to 544
Series name: guestroom
Non-Null Count  Dtype 
--------------  ----- 
545 non-null    object
dtypes: object(1)
memory usage: 27.4 KB


In [149]:
df_housing['guestroom'] = df_housing['guestroom'].astype("category")
df_housing['guestroom'].info(memory_usage="deep")

<class 'pandas.core.series.Series'>
RangeIndex: 545 entries, 0 to 544
Series name: guestroom
Non-Null Count  Dtype   
--------------  -----   
545 non-null    category
dtypes: category(1)
memory usage: 888.0 bytes


In [150]:
df_housing['basement'].info(memory_usage="deep")

<class 'pandas.core.series.Series'>
RangeIndex: 545 entries, 0 to 544
Series name: basement
Non-Null Count  Dtype 
--------------  ----- 
545 non-null    object
dtypes: object(1)
memory usage: 27.5 KB


In [151]:
df_housing['basement'] = df_housing['basement'].astype("category")
df_housing['basement'].info(memory_usage="deep")

<class 'pandas.core.series.Series'>
RangeIndex: 545 entries, 0 to 544
Series name: basement
Non-Null Count  Dtype   
--------------  -----   
545 non-null    category
dtypes: category(1)
memory usage: 888.0 bytes


In [152]:
df_housing['hotwaterheating'].info(memory_usage="deep")

<class 'pandas.core.series.Series'>
RangeIndex: 545 entries, 0 to 544
Series name: hotwaterheating
Non-Null Count  Dtype 
--------------  ----- 
545 non-null    object
dtypes: object(1)
memory usage: 27.3 KB


In [153]:
df_housing['hotwaterheating'] = df_housing['hotwaterheating'].astype("category")
df_housing['hotwaterheating'].info(memory_usage="deep")

<class 'pandas.core.series.Series'>
RangeIndex: 545 entries, 0 to 544
Series name: hotwaterheating
Non-Null Count  Dtype   
--------------  -----   
545 non-null    category
dtypes: category(1)
memory usage: 888.0 bytes


In [154]:
df_housing['airconditioning'].info(memory_usage="deep")

<class 'pandas.core.series.Series'>
RangeIndex: 545 entries, 0 to 544
Series name: airconditioning
Non-Null Count  Dtype 
--------------  ----- 
545 non-null    object
dtypes: object(1)
memory usage: 27.4 KB


In [155]:
df_housing['airconditioning'] = df_housing['airconditioning'].astype("category")
df_housing['airconditioning'].info(memory_usage="deep")

<class 'pandas.core.series.Series'>
RangeIndex: 545 entries, 0 to 544
Series name: airconditioning
Non-Null Count  Dtype   
--------------  -----   
545 non-null    category
dtypes: category(1)
memory usage: 888.0 bytes


In [156]:
df_housing['prefarea'].info(memory_usage="deep")

<class 'pandas.core.series.Series'>
RangeIndex: 545 entries, 0 to 544
Series name: prefarea
Non-Null Count  Dtype 
--------------  ----- 
545 non-null    object
dtypes: object(1)
memory usage: 27.4 KB


In [157]:
df_housing['prefarea'] = df_housing['prefarea'].astype("category")
df_housing['prefarea'].info(memory_usage="deep")

<class 'pandas.core.series.Series'>
RangeIndex: 545 entries, 0 to 544
Series name: prefarea
Non-Null Count  Dtype   
--------------  -----   
545 non-null    category
dtypes: category(1)
memory usage: 888.0 bytes


In [158]:
df_housing['furnishingstatus'].info(memory_usage="deep")

<class 'pandas.core.series.Series'>
RangeIndex: 545 entries, 0 to 544
Series name: furnishingstatus
Non-Null Count  Dtype 
--------------  ----- 
545 non-null    object
dtypes: object(1)
memory usage: 32.5 KB


In [159]:
df_housing['furnishingstatus'] = df_housing['furnishingstatus'].astype("category")
df_housing['furnishingstatus'].info(memory_usage="deep")

<class 'pandas.core.series.Series'>
RangeIndex: 545 entries, 0 to 544
Series name: furnishingstatus
Non-Null Count  Dtype   
--------------  -----   
545 non-null    category
dtypes: category(1)
memory usage: 966.0 bytes


In [160]:
df_housing.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 545 entries, 0 to 544
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype   
---  ------            --------------  -----   
 0   price             545 non-null    int8    
 1   area              545 non-null    int16   
 2   bedrooms          545 non-null    int8    
 3   bathrooms         545 non-null    int8    
 4   stories           545 non-null    int8    
 5   mainroad          545 non-null    category
 6   guestroom         545 non-null    category
 7   basement          545 non-null    category
 8   hotwaterheating   545 non-null    category
 9   airconditioning   545 non-null    category
 10  parking           545 non-null    int8    
 11  prefarea          545 non-null    category
 12  furnishingstatus  545 non-null    category
dtypes: category(7), int16(1), int8(5)
memory usage: 8.4 KB


**Результат проведенной оптимизации**

Выполненная оптимизация типов данных позволила уменьшить занимаемый объем с 55.5+ КВ до 8.4 КВ.
